Implement new Separator block options

matiasbenedetto opened the pull request:
<!-- Thanks for contributing to our free themes! Please provide as much information as possible with your Pull Request by filling out the following - this helps make reviewing much quicker! -->

#### Changes proposed in this Pull Request:
This PR implements the latest features from the Gutenberg Separator block to the following themes:

- Archeo
- ~~Blockbase~~
- Remote
- Skatepark
- Stewart

#### Related issue(s):
https://github.com/Automattic/themes/issues/5728
https://github.com/Automattic/themes/issues/5582

matiasbenedetto committed:
Remote: separator block fix

matiasbenedetto committed:
Skatepark: separator block fixes

matiasbenedetto committed:
Blockbase: separator block fixes

matiasbenedetto committed:
Seedlet: separator blocck fixes

matiasbenedetto committed:
Stewart: separator block fix

matiasbenedetto committed:
Archeo: separator block fixes

matiasbenedetto committed:
Revert "Seedlet: separator blocck fixes"

This reverts commit cca58fcee243cd49c1255169a381d74e5dd084f6.

pbking commented:
For the moment my 'druthers is to refrain from doing this in Blockbase yet...  I'm increasingly of the opinion that Blockbase should be 5.9 sans-Gutenberg compatible.

MaggieCabrera commented:
> For the moment my 'druthers is to refrain from doing this in Blockbase yet... I'm increasingly of the opinion that Blockbase should be 5.9 sans-Gutenberg compatible.

I agree that it shouldn't require GB, but why stay in 5.9 only? I think we can hold off this change until the 6.0 release, but we shouldn't stop improving Blockbase to remove things from the ponyfill like it was the original idea. We should just do it when everything is stable and in core.

pbking commented:
> I think we can hold off this change until the 6.0 release,

Yes, wholeheartedly agree.  I don't mean to stagnate Blockbase at 5.9 but rather to hold off on adding this feature to Blockbase until then.

The other themes I feel less strongly about.  I see benefit to bringing this in for the other themes today.

pbking commented on archeo/theme.json:
```diff
@@ -231,7 +231,7 @@
 			"core/separator": {
 				"color": {
-					"background": "var(--wp--preset--color--foreground)"
+					"text": "var(--wp--preset--color--foreground)"
```
I don't believe this should change to 'text'.  I believe 'background' is still the correct attribute to style.

matiasbenedetto replied:
I don't think so, please see my comments here: 
https://github.com/Automattic/themes/pull/5834#discussion_r853280069
https://github.com/Automattic/themes/pull/5834#discussion_r853212650

pbking commented on archeo/templates/single.html:
```diff
@@ -51,17 +51,23 @@
+		<div class="wp-block-column">
+		
+		<!-- wp:separator {"style":{"color":{"background":"#fcfbf766"}},"className":"is-style-wide"} -->
+		<hr class="wp-block-separator has-text-color has-alpha-channel-opacity has-background is-style-wide" style="background-color:#fcfbf766;color:#fcfbf766"/>
```
This isn't one of the palette colors.  Can it be?  I think a couple of these classes aren't needed (has-text-color has-alpha-channel-opacity) and aren't represented on the block attributes anyway.

matiasbenedetto replied:
That is the palette color named `background` (`#FCFBF8`) with the alpha channel set to 40% to preserve the same look. 

If we use `color.text` instead of `color.background` in `theme.json` we will need this change. The rationale:

We need to add that because otherwise, these separators that get their color using `currentColor` become invisible, because the block color set in `theme.json` is `foreground` (kind of black). See how `currentColor` is applied to the block in the Gutenberg codebase: https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/separator/style.scss

`currentColor` only works with `color` (`color.text` in `theme.json`) that's why is impacted by the change in theme.json and why we need to set a color in this separator.

The CSS classes are added by Gutenberg to handle the legacy separator block version. I didn't add markup manually it was just the output of the editor.

pbking commented on blockbase/functions.php:
```diff
@@ -14,6 +14,9 @@ function blockbase_support() {
 		add_theme_support( 'editor-styles' );
 
+		// Enqueue the Gutenberg plugin theme.css file
+		add_theme_support( 'wp-block-styles' );
```
Is the addition of `wp-block-styles` necessary to achieve this change?  It adds a lot of CSS to the theme and I'm hesitant to do that late in the game.  I know we evaluated it elsewhere and I believe we decided not to bring that into Blockbase.

matiasbenedetto replied:
Yep, it is necessary. But I reverted the Gutenberg styles inclusion in Blockbase to explore that change in a separated PR because, as you said, that change can have a broader impact.

pbking commented on remote/theme.json:
```diff
@@ -145,8 +145,8 @@
-				"border": {
-					"color": "var(--wp--preset--color--primary)"
+				"color": {
+					"text": "var(--wp--preset--color--primary)"
```
I think this needs to be 'background' not 'text'

matiasbenedetto replied:
I'm not sure about that, please, try it using the Remote theme as an example. If you use `background` instead of `text` you will not be able to preview the default separator color in the editor.

pbking commented:
From what I can tell the block attribute we're shifting to is `color.background` and these changes seem to target something different.

I would much prefer to keep the changes to accomplish this in Blockbase in a separate PR as the changes are potentially broader in scope, addresses different (and more) issues and we may want to time it differently (i.e. WP 6.0 release).  That said it seems to fail gracefully and it might be fine to bring it in before then.  It does fix the "three dots" issue so this is definitely the right fix for that.

I'm not entirely comfortable that this is ready, moving back to in-progress to have another pass.

matiasbenedetto committed:
Merge branch 'trunk' into update/5728

matiasbenedetto committed:
Revert "Blockbase: separator block fixes"

This reverts commit b7e0579a85b7d5b9e6721c2f1163cc68ac9aa447.

matiasbenedetto commented:
I reverted the Gutenberg styles inclusion in Blockbase to explore that change in a separated PR because that change can have a broader impact. 

The other comments from @pbking were answered in the code snippets and I think are OK. I added rationale to have a more detailed explanation of the changes. I'm moving this PR to "needs review" again.

Thanks, @MaggieCabrera @pbking for the review!

jffng commented on skatepark/inc/patterns/mixed-media-in-container.php:
```diff
@@ -69,9 +69,11 @@
-	<!-- /wp:separator --></div>
+	<div class="wp-block-column">
+	
+	<!-- wp:separator {"backgroundColor":"primary","className":"is-style-wide"} -->
```
I get a block validation error when I try to insert this pattern. The result of clicking "Attempt Block Recovery" is:

```
<!-- wp:separator {"opacity":"css","backgroundColor":"primary","className":"is-style-wide"} -->
<hr class="wp-block-separator has-text-color has-primary-color has-css-opacity has-primary-background-color has-background is-style-wide"/>
<!-- /wp:separator -->
```

jffng reviewed (commented):
Thank you for tackling this @matiasbenedetto. In general I think this would be easier to review as separate PRs per theme, especially since there are many changes to visual elements. Also there are conflicts for some reason in Skateparks patterns, any idea what's going on there?

I tested Skatepark and the results are a bit confusing.

For the Home, Search templates, all the separators become slightly thicker. I think it's okay:

Home Before | Home After
--- | ---
<img width="1241" alt="Screen Shot 2022-04-22 at 11 14 15 AM" src="https://user-images.githubusercontent.com/5375500/164743587-16d35c76-bc1e-4d21-81f8-871d2b39b26f.png"> | <img width="1259" alt="Screen Shot 2022-04-22 at 10 45 00 AM" src="https://user-images.githubusercontent.com/5375500/164740895-f348e975-ba24-4d89-87f3-616fe0f5364b.png">

Search Before | Search After
--- | ---
<img width="753" alt="Screen Shot 2022-04-22 at 10 08 18 AM" src="https://user-images.githubusercontent.com/5375500/164734914-72e7d25c-1981-4887-96cf-71c8ce15906c.png"> | <img width="802" alt="Screen Shot 2022-04-22 at 10 08 07 AM" src="https://user-images.githubusercontent.com/5375500/164734934-1ea1ec4d-98d1-4f1b-a25e-0f4e0a3db352.png">

With already existing separator content in patterns, the separators become thicker too:

Before | After
---  | ---
<img width="1229" alt="Screen Shot 2022-04-22 at 11 11 35 AM" src="https://user-images.githubusercontent.com/5375500/164743139-d44f9f6e-6890-4ae5-ae9c-989f4f84e9c7.png"> | <img width="1212" alt="Screen Shot 2022-04-22 at 11 11 23 AM" src="https://user-images.githubusercontent.com/5375500/164743400-d23c759a-00aa-4b3c-a453-c770eaa58a1e.png">

I encountered one error inserting the Mixed Media in Container pattern.

matiasbenedetto committed:
Merge branch 'trunk' into update/5728

matiasbenedetto commented:
:wave: the conflicts are related to the merge of the PR about moving the templates to a different folder. That is not strictly related to this PR, but anyway, I will create smaller PRs for each theme as @jffng proposes.

matiasbenedetto commented:
Closed in favor of https://github.com/Automattic/themes/pull/5940

matiasbenedetto commented:
Follow up for Blockbase: https://github.com/Automattic/themes/pull/5964
